src/eddymotion/utils.py

```diff
@@ -0,0 +1,176 @@
+        return random_action(dwdata, seed)
+    elif strategy == SortingStrategy.BVALUE:
+        return bvalue_action(dwdata)
+    elif strategy == SortingStrategy.CENTRALSYM:
+        return centralsym_action(dwdata)
+    else:
+        raise ValueError("Invalid sorting strategy")
+
+
+def linear_action(dwdata):
+    """
+    Sort the DWI data volume indices linearly
+
+    Parameters
+    ----------
+    dwdata : :obj:`~eddymotion.dmri.DWI` DWI dataset, represented by this tool's internal type.
+
+    Returns
+    -------
+    index_order : :obj:`numpy.ndarray`
+    The sorted index order.
+    """
+    index_order = np.arange(len(dwdata))
+
+    return index_order
+
+
+def random_action(dwdata, seed=None):
+    """Sort the DWI data volume indices.
+
+    Parameters
+    ----------
+    dwdata : :obj:`~eddymotion.dmri.DWI` DWI dataset, represented by this tool's internal type.
+    seed : :obj:`int` or :obj:`bool`, optional
+        Seed the random number generator. If an integer, the value is used to
+        initialize the generator; if ``True``, the arbitrary value
+        of ``20210324`` is used to initialize it.
+
+    Returns
+    -------
+    index_order : :obj:`numpy.ndarray`
+        The sorted index order.
+    """
+
+    _seed = None
+    if seed or seed == 0:
+        _seed = 20210324 if seed is True else seed
+
+    rng = np.random.default_rng(_seed)
+
+    index_order = np.arange(len(dwdata))
+    rng.shuffle(index_order)
+
+    return index_order
+
+
+def bvalue_action(dwdata):
+    """
+    Sort the DWI data volume indices in ascending order based on the last column of gradients.
+
+    Parameters
+    ----------
+    dwdata : :obj:`~eddymotion.dmri.DWI` DWI dataset, represented by this tool's internal type.
+
+    Returns
+    -------
+    numpy.ndarray: The sorted index order.
+    """
+    last_column = dwdata.gradients[:, -1]
+    index_order = np.argsort(last_column)
+    return index_order
+
+
+def centralsym_action(dwdata):
+    """
+    Sort the DWI data volume indices in a central symmetric manner.
+
+    Parameters
+    ----------
+    dwdata : :obj:`~eddymotion.dmri.DWI` DWI dataset, represented by this tool's internal type.
+
+    Returns
+    -------
+    numpy.ndarray: The sorted index order.
+
+    """
+    old_index = np.arange(len(dwdata))
+
+    index_order = old_index.copy()
+    if len(old_index) % 2 == 0:
+        middle_point = int(len(old_index) / 2 - 1)
+        index_order[0] = old_index[middle_point]
+
+        for i in np.arange(1, middle_point+1):
+            index_order[2 * i - 1] = old_index[middle_point + i]
+            index_order[2 * i] = old_index[middle_point - i]
+    else:
+        middle_point = int(len(old_index) / 2)
+        index_order[0] = old_index[middle_point]
+        for i in np.arange(1, middle_point + 1):
+            index_order[2 * i - 1] = old_index[middle_point + i]
+            index_order[2 * i] = old_index[middle_point - i]
+
+    return index_order
```

Comment:
```suggestion
    linear = list(range(size))
    half1, half2 = list(reversed(linear[:size // 2])), linear[size // 2:]
    index_order = [
        sub[item] for item in range(len(half1))
        for sub in [half1, half2]
    ]
    if size % 2:  # If size is odd number, append last element
        index_order.append(half2[-1])
    return index_order
```

Reply:
I don't think it's working for odd numbers, e.g., for the sequence [0, 1, 2, 3, 4, 5, 6], it returns [2, 3, 1, 4, 0, 5, 6]. From our discussion yesterday, I was expecting either [3, 4, 2, 5, 1, 6, 0] or [3, 2, 4, 1, 5, 0, 6], but perhaps I misunderstood. Could you confirm whether we should expect [3, 4, 2, 5, 1, 6, 0] or [3, 2, 4, 1, 5, 0, 6]?

Reply:
Yes sure. I agree with the expected solution, and I committed the suggestion to confirm there is an issue with the implementation of the method.

Reply:
> 

Before we merge, there's still this issue with the implementation of the method. Would you be ok with me making modifications to your code, or would you prefer to handle the changes yourself?